planner/core/find_best_task.go

```diff
@@ -249,6 +249,14 @@ func (p *baseLogicalPlan) enumeratePhysicalPlans4Task(physicalPlans []PhysicalPl
 			}
 		}
 
+		// An optimal task could not satisfy the property, so it should be converted here.
+		if prop.TaskTp == property.RootTaskType {
```

Comment:
@hanfei1991 here we should enforce this change, when this operator can be pushed down, like broadcast join.